modAionImpl/src/org/aion/zero/impl/AionBlockchainImpl.java

```diff
@@ -1170,19 +1171,18 @@ private AionBlockSummary applyBlock(IAionBlock block) {
     }
 
     /**
-     * Returns a {@link PostExecutionWork} object whose {@code doPostExecutionWork()} method will run
-     * the provided logic defined in this method. This work is to be applied after each transaction
-     * has been run.
+     * Returns a {@link PostExecutionWork} object whose {@code doPostExecutionWork()} method will
+     * run the provided logic defined in this method. This work is to be applied after each
+     * transaction has been run.
      *
-     * This "work" is specific to the {@link AionBlockchainImpl#applyBlock(IAionBlock)} method.
+     * <p>This "work" is specific to the {@link AionBlockchainImpl#applyBlock(IAionBlock)} method.
      */
     private static PostExecutionWork getPostExecutionWorkForApplyBlock() {
         return (topRepository,
-            childRepository,
-            transactionSummary,
-            transaction,
-            blockEnergyLeft) -> {
-
+                childRepository,
```

Comment:
since the lambda function doesn't need to change at each call of this method, can do something like this to instantiate it once only:

```java
    private static PostExecutionWork PEW = (topRepository, childRepository, transactionSummary, transaction, blockEnergyRemaining) -> {
        if (!transactionSummary.isRejected()) {
            childRepository.flush();

            AionTxReceipt receipt = transactionSummary.getReceipt();
            receipt.setPostTxState(topRepository.getRoot());
            receipt.setTransaction(transaction);

            return receipt.getEnergyUsed();
        } else {
            return 0;
        }
    };

    private static PostExecutionWork getPostExecutionWorkForGeneratePreBlock2 () {
        return PEW;
    }
```

Reply:
I'm going to also move this into the next set of refactorings, because I don't want to alter logic at this point, there's pressure to get these changes in